### ISSUE TITLE
UNAIDS 2015 expenditure data 

### ISSUE THREAD

**Issue by bezunakarmi** [devinit/D-Portal#374]
Expenditure amount for[ UNIAIDS for the year 2015](http://d-portal.org/ctrack.html?publisher=XM-DAC-41110&tongue=eng#view=publisher_countries) is not shown in the portal. While viewing the data in IATI registry ([here](http://preview.iatistandard.org/index.php?url=https%3A//aidstream.org/files/xml/unaids-998.xml)) , we can see that there are transactions flowing to multiple recipient countries.

We have been using d-portal's code for our [system](http://open.undg.org/). The issue that we are facing is while aggregating the transaction amount for multiple recipient countries with in a single activity(in above mentioned publisher), all the country show same total of transaction amount. 

Could you verify this issue?

**Comment by Mattdevinit** [comment]
thanks @bezunakarmi 
think this is due to the structure in the data which is correct but different to most other publishers in that the recipient country is at transaction level rather than activity level.  As this is the first time we've come across it, it needs a change in the d-portal code to look for recipient country (or region) at transaction level if not present at activity level, and this should mean the data would show up ok in this instance.
Also, would be worth updating the UN portal code at some point too, as it's running on an older version

**Comment by Mattdevinit** [comment]
@bezunakarmi quick update - we're aiming to finish current work on the search function updates in the next 4 weeks and then will address this issue of both sectors and recipient countries at transaction level, so we hope to have a solution live within August

**Comment by JamesCoe** [comment]
Hi @Mattdevinit -- is there any update on this?

I noticed that USAID's data is not displaying on d-Portal when you filter by sector. When, for example, I search for all USAID activities in Tanzania I get 2472 total projects. However, when I filter by sectors I consistently get 0 results. 

I wonder if this is being caused by a similar issue as described above, i.e. USAID uses DAC at the transaction level, rather than activity and so d-Portal isn't picking them up when it filters?

**Comment by Mattdevinit** [comment]
Hi @JamesCoe - thanks for following up, we should be done with Search updates by the end of this week (after a short delay), and next priority after that is addressing countries/sectors at transaction level.  I think you're right, that this is the reason the USAID sector data isn't showing so hopefully when we address the data at transaction level this will fix it.

**Comment by JamesCoe** [comment]
@Mattdevinit -- Sounds good, cheers!

**Comment by JamesCoe** [comment]
Hi @Mattdevinit -- is there any update on the ETC for the issue re: countries/sectors at transaction level.

**Comment by Mattdevinit** [comment]
Hi @JamesCoe - we've started work on this now, following completion of search updates.  It's more complex than we'd anticipated as it involves comprehensive restructuring of the core of d-portal, so is taking some careful work!  We're hoping that this will be in place by the end of this month - I'll add an update here as this progresses - thanks

**Comment by JamesCoe** [comment]
Hi @Mattdevinit -- no worries and cheers! 

I am hoping to pull together a 'data use' workshop in Washington DC, so I can show some people what IATI is about and how it can be used. If I drop you a line next week, would it be possible to please let me know if things are still on track to be done by end of Oct? I'm hesitant to do a workshop on using IATI etc if the US' data doesn't display.

**Comment by Mattdevinit** [comment]
hi @JamesCoe - yes of course please do.  I'll aim to put some updates here too (although am on leave last week of Sept).  It's end of September we are aiming for - do you have an idea when the data use workshop will be?

**Comment by JamesCoe** [comment]
Hi @Mattdevinit -- perfect, cheers! And yep, we're aiming for something around 18 October.

**Comment by notshi** [comment]
Hi @JamesCoe - We've whittled this down to a two-part process. We are currently at 1.

1. The database is currently being reconfigured on the live server to add this extra information to ```<transaction>``` and ```<budget>```

While this is happening, d-portal updates are currently turned off so that we don't get any conflicts during this period. We are trying hard to avoid downtime. **If anything goes wrong whilst we are doing this - it will take 3 days to repopulate the database.**

Once this is done, we'll have to check to make sure that we haven't lost any numbers in this new import process.

2. Switch the client-side over to display this new extra information 

This involves how d-portal queries for sector and country information as this data is now also stored inside ```<transaction>``` and ```<budget>```.

**Old code should continue to work exactly as is but the new way will obviously be more precise.**

Finally we'll need to work out how we associate an activity with countries or sectors as it's entirely possible that sectors and countries are listed within transaction data but are not listed in the main activity; ie.  activities reporting ```<recipient-country>``` with percentages but a different country inside ```<transaction>```.

This will then trickle down to how these new child elements get displayed within SAVi and if they should also affect the sector graphs on those pages.

**Basically, this is a huge can of worms and not quite as simple as it seems because we have to deal with the two ways of reporting the same data simultaneously.**

**Comment by JamesCoe** [comment]
Hi @notshi -- sorry for the delay, this email went into my junk inbox! Cheers for updating me, I appreciate it. Sorry to hear that this is a bit more complicated than expected! Do you still think its possible to get this issue resolved by 19 October?

**Comment by Mattdevinit** [comment]
Hi @JamesCoe I think from discussions with Kriss and Shi and looking at the complexity involved on a number of areas (which are becoming more apparent as we work on this) that finishing this by 19 Oct is unlikely (but not impossible).  We'll update again next week, and worth noting that there's been a pause whilst the Member's Assembly has been going on to ensure nothing broke during that time!

**Comment by Mattdevinit** [comment]
@JamesCoe as you've seen with communication from Bill some data is now available on ths in d-portal - just a heads up that there's still further work ongoing up til early next week, so there will be some discrepancies in the data until all of this development is complete

**Comment by JamesCoe** [comment]
@Mattdevinit Cheers! I appreciate it.

**Comment by notshi** [comment]
Thanks, everyone for your patience. Am closing this as there is data populating the tables.
Please open a new issue if otherwise.
